src/Kestrel.Core/Internal/Http2/Http2FrameWriter.cs

```diff
@@ -109,14 +113,34 @@ public Task Write100ContinueAsync(int streamId)
         {
             lock (_writeLock)
             {
+                if (_completed)
+                {
+                    return Task.CompletedTask;
+                }
+
                 _outgoingFrame.PrepareHeaders(Http2HeadersFrameFlags.END_HEADERS, streamId);
                 _outgoingFrame.PayloadLength = _continueBytes.Length;
-                _continueBytes.CopyTo(_outgoingFrame.HeadersPayload);
-
-                return WriteFrameUnsynchronizedAsync();
+                WriteHeaderUnsynchronized();
+                _outputWriter.Write(_continueBytes);
+                return _flusher.FlushAsync();
             }
         }
 
+        // TODO: Why doesn't this method flush? Is that because we always write a data frame afterwards (even if it's empty)?
```

Comment:
I would assume so, but is there a penalty to flushing early?

Reply:
Flushing early creates more TLS & TCP packets.